### PR TITLE
Changing to cold wallet and adding tests

### PR DESCRIPTION
**Pull request by teghnet**

### CODE REVIEW

File: app/Utilities/DatabaseConfigReader.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ public static function overwriteAdministrationConfig(): void
         $settings = Config::fetchAdminSettings();
         $hotWalletMinValue = $settings[Config::HOT_WALLET_MIN_VALUE];
         $hotWalletMaxValue = $settings[Config::HOT_WALLET_MAX_VALUE];
-        $hotWalletAddress = $settings[Config::HOT_WALLET_ADDRESS] ?? '';
+        $hotWalletAddress = $settings[Config::COLD_WALLET_ADDRESS] ?? '';
```
Comment (PawelPodkalicki) [inline app/Utilities/DatabaseConfigReader.php]: Rename variable

File: app/Models/Config.php [hunk 2]
```diff
@@ -162,7 +166,7 @@ public static function fetchAdminSettings(): array
 
     public static function isHotWalletActive(): bool
```
Comment (PawelPodkalicki) [inline app/Models/Config.php]: Rename function

File: database/migrations/2019_04_05_072801_insert_config_adpay_event_last_exported_id.php [hunk 3]
```diff
@@ -46,6 +46,8 @@ public function up(): void
         DB::table('configs')->updateOrInsert(
             [
                 'key' => Config::ADPAY_LAST_EXPORTED_EVENT_ID,
+            ],
```
Comment (PawelPodkalicki) [inline database/migrations/2019_04_05_072801_insert_config_adpay_event_last_exported_id.php:49]: Was that tested and is it really needed?

File: database/migrations/2019_04_11_084433_update_config_entry_name_for_cold_wallet.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            } else {
+                $hotWalletIsActive->key = Config::COLD_WALLET_IS_ACTIVE;
+                $hotWalletIsActive->save();
+            }
+        }
+
+        $coldWalletAddress = DB::table('configs')->where('key', Config::COLD_WALLET_ADDRESS)->first();
+        if ($coldWalletAddress === null) {
+            $hotWalletAddress = DB::table('configs')->where('key', 'hotwallet-address')->first();
+            if ($hotWalletAddress === null && $coldWalletAddress === null) {
```
Comment (PawelPodkalicki) [inline database/migrations/2019_04_11_084433_update_config_entry_name_for_cold_wallet.php]: Condition `$coldWalletAddress === null` is checked above.
```suggestion
            if ($hotWalletAddress === null) {
```

**Review by PawelPodkalicki — CHANGES_REQUESTED** [review]

File: database/migrations/2019_04_05_072801_insert_config_adpay_event_last_exported_id.php (hunk 3 above)
Comment (teghnet) [reply database/migrations/2019_04_05_072801_insert_config_adpay_event_last_exported_id.php:49]: yes

**Review by PawelPodkalicki — APPROVED** [review]

**Pull request merged**